### DISCUSSION TITLE
Zustand with a Database like Supabase

### DISCUSSION THREAD

**Discussion by mawnir** [pmndrs/zustand#2284] [Q&A]
I'm working on an app like Notion, i need to update the "Fav list", "Main list" or the "trash list" based on certain actions in the opened document. i can achieve this connecting directly to Supabase, but what i need is to improve speed and have localstorage/cache... Zustand can help with both.
My question is what is the best approach to do this? is it to connect to whole Table to the store and then do actions locally and then update the DB? or simple stores with actions like 'isFavorite' ==> true...? any example will be great
![Untitled-2023-04-18-2356](https://github.com/pmndrs/zustand/assets/62011430/042681c5-eb60-4e13-823e-72c20c2a3636)

**Comment by dbritto-dev** [comment]
@mawnir well in this case since you need to persist your changes I would recommend use `persist` middleware with custom storage so you can sync your changes with supabase db
